### DISCUSSION TITLE
Flow Switch Options

### DISCUSSION THREAD

**Discussion by Telemachus1977** [tagyoureit/nodejs-poolController#625] [General]
I helped my buddy get his REM Chemical controller up and running today, alongside his ET4.  Running fine on a raspberry Pi 0 W with the four relay four input card.

We built his flow cell with one inch PVC that sits on top of his solar circuit with 3D printed saddle brackets.

Everything seems to be working great, but the 3/8 in tubing that feeds this flow cell creates too much of a bottleneck for the flow meter that we got.  It's made for a spa, which obviously has a lot more flow than what we are getting through this circuit.

We tried all sorts of modifications for it, but couldn't find anything that would consistently monitor flow with different pump speeds.

Any other ideas for what else we could use in that three quarter inch opening?

Otherwise we have found some flow sensors that looks like they would fit right in the 3/8 in tubing and we could just cap this opening.

![20221103_143844](https://user-images.githubusercontent.com/25725743/199875507-27b89e5f-1fa7-48fb-8d24-ff2a9fb80c4b.jpg)
![20221103_143852](https://user-images.githubusercontent.com/25725743/199875509-d62db6f8-645a-41f5-bc85-087131d506bc.jpg)
![20221103_153537](https://user-images.githubusercontent.com/25725743/199875511-1eff92d3-aeda-4be1-b228-28f69886101e.jpg)

**Comment by rstrouse** [comment]
Awesome work.  Damn 3d printers.   Stenner makes one that triggers at 1.2 gpm.  The other option is to install a pressure transducer in the filter relief valve.

**Reply by Telemachus1977** [reply]
Oh yeah, by the way. You totally need a 3D printer.

Found another one that is 0.5 gallons per minute, but much more than he wants to spend.

https://thomasprod.com/product/2650-series-liquid-flow-switch/?attribute_flow-set-point=0.5+Increasing+flow&attribute_conduit-connector=Conduit+Connector&dTribesID=f1285f393d701e1ca39936dc841d9ffc%7Cadtribes%7C6820&utm_source=Google%20Shopping&utm_campaign=ThomasProd%20Feed&utm_medium=cpc&utm_term=6820&gclid=CjwKCAjwzY2bBhB6EiwAPpUpZi0SGQOJLYuJZ23FXRyXfyz3OaaRZqNv2jF0Ul9Lg-m4I8lrPIyyJBoCcK0QAvD_BwE

Just out of curiosity, is it not possible to tap into the flow sensor on the intellichlor?

**Reply by tagyoureit** [reply]
Looks great. 

First, you should be able to tap into the Intellichlor flow switch. See the wiring diagram here.  https://www.troublefreepool.com/wiki/index.php?title=Pentair_Intellichlor_ICXX_SWG

Second thought... make your own?  You could try to print a Thing with the 3-D printer. https://www.thingiverse.com/thing:4697095

Also, if there is any chance of reaching freezing temperatures make sure that the Atlas sensors are protected like your lemon or orange trees and are not allowed to freeze. This will destroy them.

**Reply by Telemachus1977** [reply]
Looking at videos of people replacing the flow switch on the intellichlor, it certainly would be doable.  It would involve cutting the wiring and rewiring the flow switch.  Probably not a avenue that most people would want to pursue given the cost of these SWGs nowadays.

Although the Inland Empire of Southern California rarely gets to freezing, it does happen occasionally.

My solution was to take some extra insulated foam that I had and I built a ugly igloo that goes around my entire flow cell.  Hopefully some of the heat from the panel will transmit through the metal and keep it all nice and cozy.  I remove my salt cell every winter anyways, so I will probably pull the orp probe.

![20221104_073002](https://user-images.githubusercontent.com/25725743/200017646-4da51a97-88e0-4ded-bde9-49b5ba9073fd.jpg)
![20221104_072957](https://user-images.githubusercontent.com/25725743/200017657-279e3fad-93f3-448b-a7fb-8887c0815253.jpg)
![20221104_072932](https://user-images.githubusercontent.com/25725743/200017666-51cdce06-bf84-4b1a-994e-923b19647006.jpg)


As for my buddy's industrial probes, would putting some of that black pool noodle pipe insulation over them be enough? Or do you think that the entire flow circuit should be wrapped?

**Reply by rstrouse** [reply]
Your shield should be plenty with the heat from the transformers even through a pretty extended freeze.  For your buddy the pool noodle will probably be fine for overnight frost.  If you get a hard freeze I would probably take more measures.  Maybe throw a tarp over the probes that includes the pump.  The pump should run whenever 35F is found.  The bottom of the probes are in a pretty good insulator when the water is running but the top part where the standard solution is stored should not be allowed to freeze.

**Reply by Telemachus1977** [reply]
![Screenshot_20221104_085647_VNC Viewer](https://user-images.githubusercontent.com/25725743/200032910-7895d093-54d8-4c7c-94d9-971ac35efba1.jpg)

His date and time are off by about 24 hours. We have tried using server time and it didn't change. Currently it is set to manual time.

Any suggestions on why the time is off?

**Reply by tagyoureit** [reply]
Use server time and wait at least 30 mins.  It will check the ET time agains the RPi time at :30 and :00 past the hour.  It will update if it is >2 mins difference.  If it doesn't update, make sure the RPi time and timezone are properly set.  

Using manual time you need to update directly on ET panel or use the API.  We don't have a place to set the ET time manually in dP.

**Reply by Telemachus1977** [reply]
That's the strange thing, on manual it is showing a different time than the Easy Touch panel.

I'll have him check and change the raspberry Pi time.

**Reply by Telemachus1977** [reply]
Several cheap hall sensors like this on Amazon

DIGITEN G3/8 Inch Quick Connect Water Flow Sensor Switch Flowmeter Counter 0.3-10L/min https://a.co/d/7UWwD15

It looks like they'll need some protection from the elements, but fairly simple to add in line with the poly tubing.

Do you think the square wave output of the hall signal wire would be read as a steady voltage input for the sequent 4relay 4 input board?

**Reply by rstrouse** [reply]
I would opt for pressure on the filter air bleed as this is cheap and gives you more information that can be used to determine how dirty the filter is.

**Reply by rstrouse** [reply]
I didn't really answer your question.  I do not believe you can look at a pulse count and determine instantaneous flow.   The real way to deal with a metering device like that is to couple it with a microcontroller with interrupts and sample a range.  I have thought about building a Trinket M0 or Feather that can read sensors like that.

**Reply by Telemachus1977** [reply]
I'm not really thinking about measuring the pulse count, but more the resulting voltage.

If I am feeding 5 volts to the input and the hall signal is returning a pulse signal that results to something around 3 or 4 v, I wonder if that would registered as a positive voltage.  

The input on this board is optically isolated, so likely just looking for an LED light lighting up?

**Reply by Telemachus1977** [reply]
What hardware would you use to monitor the pressure on the filter?  He doesn't have a mega-BAS

EZO-PRS looks like it could plug right into one of the extra ports on the tentacle t3, but expensive.

Otherwise, he could use a generic pressure transducer with a standard ADC board, right?

**Reply by tagyoureit** [reply]
You would use a PRS like https://www.ebay.com/itm/284015552963.  It would output 0.5-5v dc that you would read with the input on the Sequent board.

**Reply by Telemachus1977** [reply]
Cool thank you for the link.

The sequent 4 relay input board does not have any sort of adc, but I'm sure it would work for a simple on-off.

We also have one of these that we know works with REM.

![43871](https://user-images.githubusercontent.com/25725743/200141485-4a6c5262-bd8a-4312-998a-95267914d79d.jpeg)


I assume the pressure transducer will work okay with 3.3 volts, but after reading about Hect0r's experience in not so sure now.
